Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+        /// </returns>
+        public bool Equals(EmailAddressLocalPart other)
+        {
+            if (other == null)
+                return false; 
+
+            return this.localPart.Equals(other.localPart);
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object"/> is equal to this instance.
+        /// </summary>
+        /// <param name="obj">The <see cref="System.Object"/> to compare with this instance.
+        /// </param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object"/> is equal to this instance; 
+        /// otherwise, <c>false</c>.
+        /// </returns>
+        /// <exception cref="T:System.NullReferenceException">
+        /// The <paramref name="obj"/> parameter is null.
+        ///   </exception>
+        public override bool Equals(object obj)
+        {
+            var other = obj as EmailAddressLocalPart;
+            if (other != null)
+            {
+                return this.Equals(other);
+            }
+
+            return base.Equals(obj);
+        }
+
+        /// <summary>
+        /// Returns a hash code for this instance.
+        /// </summary>
+        /// <returns>
+        /// A hash code for this instance, suitable for use in hashing algorithms and data 
+        /// structures like a hash table. 
+        /// </returns>
+        public override int GetHashCode()
+        {
+            return this.localPart.GetHashCode();
+        }
+
+        /// <summary>
+        /// Returns a <see cref="System.String"/> that represents the local part for this instance.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="System.String"/> that represents the local part for this instance.
+        /// </returns>
+        public override string ToString()
+        {
+            return this.localPart;
+        }       
+
+        /// <summary>
+        /// Returns true if the localPart is valid in terms of both length and conformity with allowable
+        /// characters and character ordering, false otherwise. 
+        /// </summary>
+        /// <param name="localPart"></param>
+        /// <returns></returns>
+        public static bool IsValid(string localPart)
+        {
+            if (string.IsNullOrEmpty(localPart))
+                return false;
+
+            return localPart.Length <= MaximumAllowableLength && Regex.IsMatch(localPart, ValidEmailAddressPattern);
+        }
+
+        /// <summary>
+        /// Gets the maximum length, inclusive, supported as the local part. 
+        /// </summary>
+        public static int MaximumAllowableLength
+        {
+            get { return 64; }
+        }
+
+        /// <summary>
+        /// Gets the pattern used by EmailAddressLocalPart to validate local parts.  This pattern is used
+        /// to validate for character content, but not for length.
+        /// </summary>
+        public static string ValidEmailAddressPattern
```

Comment:
Isn't this an implementation detail?

What if one day we'd like to refactor this implementation to use parser combinators instead of regular expressions?